src/Tinter.js

```diff
@@ -0,0 +1,214 @@
+
+var keyHex = [
+    "#76CFA6",
+    "#EAF5F0",
+    "#D3EFE1",
+];
+
+// cache of our replacement colours
+// defaults to our keys.
+var colors = [
+    keyHex[0],
+    keyHex[1],
+    keyHex[2],
+];
+
+var cssFixups = [
+    // {
+    //     style: a style object that should be fixed up taken from a stylesheet
+    //     attr: name of the attribute to be clobbered, e.g. 'color'
+    //     index: ordinal of primary, secondary or tertiary
+    // }
+];
+
+// CSS attributes to be fixed up
+var cssAttrs = [
+    "color",
+    "backgroundColor",
+    "borderColor",
+];
+
+var svgFixups = [
+    // {
+    //     node: a SVG node that needs to be fixed up
+    //     attr: name of the attribute to be clobbered, e.g. 'fill'
+    //     index: ordinal of primary, secondary
+    // }
+];
+
+var svgAttrs = [
+    "fill",
+    "stroke",
+];
+
+var svgNodes = {};
+
+var cached = false;
+
+function calcCssFixups() {
+    for (var i = 0; i < document.styleSheets.length; i++) {
+        var ss = document.styleSheets[i];
+        for (var j = 0; j < ss.cssRules.length; j++) {
+            var rule = ss.cssRules[j];
+            for (var k = 0; k < cssAttrs.length; k++) {
+                var attr = cssAttrs[k];
+                for (var l = 0; l < keyRgb.length; l++) {
+                    if (rule.style && rule.style[attr] === keyRgb[l]) {
+                        cssFixups.push({
+                            style: rule.style,
+                            attr: attr,
+                            index: l,
+                        });
+                    }
+                }
+            }
+        }
+    }
+}
+
+function calcSvgFixups(nodes) {
+    var svgs = nodes || document.getElementsByClassName("mx_Svg");
+    var fixups = [];
+    for (var i = 0; i < svgs.length; i++) {
+
+        var svgDoc = svgs[i].contentDocument;
+        if (!svgDoc) continue;
+        var tags = svgDoc.getElementsByTagName("*");
+        for (var j = 0; j < tags.length; j++) {
+            var tag = tags[j];
+            for (var k = 0; k < svgAttrs.length; k++) {
+                var attr = svgAttrs[k];
+                for (var l = 0; l < keyHex.length; l++) {
+                    if (tag.getAttribute(attr) && tag.getAttribute(attr).toUpperCase() === keyHex[l]) {
+                        fixups.push({
+                            node: tag,
+                            attr: attr,
+                            index: l,
+                        });
+                    }
+                }
+            }
+        }
+    }
+
+    return fixups;
+}
+
+function applyCssFixups() {
+    for (var i = 0; i < cssFixups.length; i++) {
+        var cssFixup = cssFixups[i];
+        cssFixup.style[cssFixup.attr] = colors[cssFixup.index];
+    }
+}
+
+function applySvgFixups(fixups) {
+    for (var i = 0; i < fixups.length; i++) {
+        var svgFixup = fixups[i];
+        svgFixup.node.setAttribute(svgFixup.attr, colors[svgFixup.index]);
+    }
+}
+
+function _onAction(payload) {
```

Comment:
Also, as it stands, does this not race? Every tintable thing fires off a `svg_onload` event, and we don't seem to specify which things we want to tint anywhere so this will be tinting ALL THE THINGS! The only reason why it doesn't is because it doesn't get `svg_onload` events fired at it all the time?